third_party/ibis/ibis_oracle/client.py

```diff
@@ -12,21 +12,60 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-
-
 import contextlib
 import getpass
 import os
 from typing import Optional
-
 import sqlalchemy as sa
+
+from sqlalchemy.dialects.oracle.cx_oracle import OracleDialect_cx_oracle
 from third_party.ibis.ibis_oracle.compiler import OracleDialect
 
+import ibis.expr.datatypes as dt
 import ibis.backends.base_sqlalchemy.alchemy as alch
-
 import cx_Oracle  # NOQA fail early if the driver is missing
 
 
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.CLOB)
+def sa_oracle_CLOB(_, satype, nullable=True):
+    return dt.String(nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.NCLOB)
+def sa_oracle_NCLOB(_, satype, nullable=True):
+    return dt.String(nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.LONG)
+def sa_oracle_LONG(_, satype, nullable=True):
+    return dt11.String(nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.NUMBER)
+def sa_oracle_NUMBER(_, satype, nullable=True):
+    return dt.Decimal(satype.precision, satype.scale, nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.BFILE)
+def sa_oracle_BFILE(_, satype, nullable=True):
+    return dt.Binary(nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.RAW)
+def sa_oracle_RAW(_, satype, nullable=True):
+    return dt.Binary(nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.DATE)
+def sa_oracle_DATE(_, satype, nullable=True):
+    return dt.Date(nullable=nullable)
+
+
+@dt.dtype.register(OracleDialect_cx_oracle, sa.dialects.oracle.VARCHAR)
+def sa_oracle_VARCHAR(_, satype, nullable=True):
+    return dt.String(nullable=nullable)
+
```

Comment:
@nehanene15  - Will it be a good idea to include all the below data types too or whatever you have here is sufficient?

from sqlalchemy.dialects.oracle import \
            BFILE, BLOB, CHAR, CLOB, DATE, \
            DOUBLE_PRECISION, FLOAT, INTERVAL, LONG, NCLOB, NCHAR, \
            NUMBER, NVARCHAR, NVARCHAR2, RAW, TIMESTAMP, VARCHAR, \
            VARCHAR2
https://docs.sqlalchemy.org/en/14/dialects/oracle.html#oracle-data-types

Reply:
Good point - I'll add TIMESTAMP and VARCHAR2 since those will probably be the most common